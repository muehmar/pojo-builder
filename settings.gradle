rootProject.name = "pojo-builder"

include 'core'
include 'core-annotations'
include 'example'
include 'example-java17'

dependencyResolutionManagement {
    versionCatalogs {
        libs {
            version('pojobuilder', '1.6.0')
            library('pojobuilder-core', 'io.github.muehmar','pojo-builder').versionRef('pojobuilder')
            library('pojobuilder-annotations', 'io.github.muehmar','pojo-builder-annotations').versionRef('pojobuilder')

            library('codeGenerator', 'io.github.muehmar:code-generator:0.22.0')

            version('junit', '5.10.0')
            library('junit-api', 'org.junit.jupiter', 'junit-jupiter-api').versionRef('junit')
            library('junit-engine', 'org.junit.jupiter', 'junit-jupiter-engine').versionRef('junit')
            library('junit-params', 'org.junit.jupiter', 'junit-jupiter-params').versionRef('junit')

            version('assertj', '3.26.3')
            library('assertj', 'org.assertj', 'assertj-core').versionRef('assertj')

            bundle('junit', ['junit-api', 'junit-engine', 'junit-params', 'assertj'])

            library('jakarta-annotationApi', 'jakarta.annotation:jakarta.annotation-api:2.1.1')

            library('google-autoservice', 'com.google.auto.service:auto-service:1.1.1')
            library('joor', 'org.jooq:joor-java-8:0.9.15')

            library('lombok', 'org.projectlombok:lombok:1.18.32')

            library('snapshotTesting','io.github.origin-energy:java-snapshot-testing-junit5:4.0.8')

            library('slf4j', 'org.slf4j:slf4j-api:2.0.7')

            plugin('release', 'net.researchgate.release').version('2.8.1')
            plugin('versions', 'com.github.ben-manes.versions').version('0.51.0')
        }
    }
}