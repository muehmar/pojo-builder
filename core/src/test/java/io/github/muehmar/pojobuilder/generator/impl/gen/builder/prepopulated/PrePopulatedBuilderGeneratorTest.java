package io.github.muehmar.pojobuilder.generator.impl.gen.builder.prepopulated;

import static io.github.muehmar.codegenerator.writer.Writer.javaWriter;
import static io.github.muehmar.pojobuilder.generator.model.settings.PojoSettings.defaultSettings;
import static io.github.muehmar.pojobuilder.snapshottesting.SnapshotUtil.writerSnapshot;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.annotations.SnapshotName;
import io.github.muehmar.codegenerator.Generator;
import io.github.muehmar.codegenerator.writer.Writer;
import io.github.muehmar.pojobuilder.generator.Pojos;
import io.github.muehmar.pojobuilder.generator.model.Pojo;
import io.github.muehmar.pojobuilder.generator.model.settings.PojoSettings;
import io.github.muehmar.pojobuilder.snapshottesting.SnapshotTest;
import org.junit.jupiter.api.Test;

@SnapshotTest
class PrePopulatedBuilderGeneratorTest {
  private Expect expect;

  @Test
  @SnapshotName("genericPojo")
  void prePopulatedBuilderGenerator_when_genericPojo_then_correctOutput() {
    final Generator<Pojo, PojoSettings> gen =
        PrePopulatedBuilderGenerator.prePopulatedBuilderGenerator();

    final Writer writer = gen.generate(Pojos.genericSample(), defaultSettings(), javaWriter());

    expect.toMatchSnapshot(writerSnapshot(writer));
  }

  @Test
  @SnapshotName("samplePojo")
  void prePopulatedBuilderGenerator_when_samplePojo_then_correctOutput() {
    final Generator<Pojo, PojoSettings> gen =
        PrePopulatedBuilderGenerator.prePopulatedBuilderGenerator();

    final Writer writer = gen.generate(Pojos.sample(), defaultSettings(), javaWriter());

    expect.toMatchSnapshot(writerSnapshot(writer));
  }
}
