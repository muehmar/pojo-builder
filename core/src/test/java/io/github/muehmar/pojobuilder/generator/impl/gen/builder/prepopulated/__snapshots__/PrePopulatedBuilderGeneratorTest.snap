genericPojo=[
java.io.IOException
java.lang.String
java.util.Optional

private String id;
private T data;
private S additionalData;

private CustomerBuilder(String id, T data, S additionalData) {
  this.id = id;
  this.data = data;
  this.additionalData = additionalData;
}

public CustomerBuilder<T, S> id(String id) {
  this.id = id;
  return this;
}

public CustomerBuilder<T, S> data(T data) {
  this.data = data;
  return this;
}

public CustomerBuilder<T, S> additionalData(S additionalData) {
  this.additionalData = additionalData;
  return this;
}

public CustomerBuilder<T, S> additionalData(Optional<S> additionalData) {
  this.additionalData = additionalData.orElse(null);
  return this;
}

public Customer<T, S> build() throws IOException {
  final Customer<T, S> instance =
      new Customer<T, S>(id, data, additionalData);
  return instance;
}
]


samplePojo=[
java.io.IOException
java.lang.Integer
java.lang.String
java.util.Optional

private Integer id;
private String username;
private String nickname;

private CustomerBuilder(Integer id, String username, String nickname) {
  this.id = id;
  this.username = username;
  this.nickname = nickname;
}

public CustomerBuilder id(Integer id) {
  this.id = id;
  return this;
}

public CustomerBuilder username(String username) {
  this.username = username;
  return this;
}

public CustomerBuilder nickname(String nickname) {
  this.nickname = nickname;
  return this;
}

public CustomerBuilder nickname(Optional<String> nickname) {
  this.nickname = nickname.orElse(null);
  return this;
}

public Customer build() throws IOException {
  final Customer instance =
      new Customer(id, username, nickname);
  return instance;
}
]